### PR TITLE
Add virtual request type as option on Add Hearing Day modal

### PR DESCRIPTION
**Pull request by rubaiyat22**
Resolves https://vajira.max.gov/browse/CASEFLOW-388

### Description
- Add logic to `HearingDayAddModal` component to support `Virtual` as a request type as well as refactor the component into a functional component
- Add logic to `HearingDayEditModal` to disable `Change Room` checkbox if hearing request type is `Virtual`
- Write feature tests

### Testing Plan
1. Login as user `BVASYELLOW`

#### With Feature Toggle enabled
2. in Console run  `FeatureToggle.enable!(:national_vh_queue, users: ["BVASYELLOW"])`
##### Add Hearing Day Modal
3. Go to http://localhost:3000/hearings/schedule
4. Click `Add Hearing Date` button and ensure that a modal pops open
5. Expect to see `Virtual` as a option under `Select Hearing Type` dropdown
6. Click on `Video` and ensure that `Virtual Hearings` doe not show up as a option `Select Regional Office (RO)` dropdown
6. Click on `Virtual` and expect to see dropdown `Select Regional Office (RO)` to be disabled with `Virtual Hearings` selected
7. Expect to see checkbox `Assign Board Hearing Room` to be disabled
8. Choose a date and confirm 
9. Expect to see a hearing day created with Virtual as request type and no room
##### Edit Hearing Day modal
3. Go to http://localhost:3000/hearings/schedule
4. Pick a virtual hearing day and go to daily docket page
5. Click on `Edit Hearing Day` on top left corner and expect to see modal
6. Expect to see `Change Room` checkbox to be disabled
7. Make updates to Judge, coordinator, or notes and ensure that values are saved successfully

#### Without Feature Toggle enabled
2. in Console run `FeatureToggle.disable!(:national_vh_queue, users: ["BVASYELLOW"])`
##### Add Hearing Day Modal
3. Go to http://localhost:3000/hearings/schedule
4. Click `Add Hearing Date` button and ensure that a modal pops open
5. Expect to see only `Video` and `Central` as a options under `Select Hearing Type` dropdown
6.  Click on `Video` and ensure that `Virtual Hearings` doe not show up as a option `Select Regional Office (RO)` dropdown
6. Test to ensure that you can create video and central as before
##### Edit Hearing Day modal
3. Go to http://localhost:3000/hearings/schedule
4. Pick a video/central hearing day and go to daily docket page
5. Click on `Edit Hearing Day` on top left corner and expect to see modal
6. Ensure that all checkboxes are enabled and you can make edits to all fields

#### Add Hearing Day Modal
Screen Shot 2021-01-21 at 12 09 09 PM

#### Edit Hearing Day Modal
Screen Shot 2021-01-21 at 12 08 22 PM

### CODE REVIEW

File: app/views/hearings/index.html.erb [hunk 1]
```diff
@@ -17,6 +17,7 @@
   userUseFullPageVideoToVirtual: FeatureToggle.enabled?(:full_page_video_to_virtual, user: current_user),
   userCanScheduleVirtualHearings: FeatureToggle.enabled?(:schedule_virtual_hearings, user: current_user),
   userCanConvertCentralHearings: FeatureToggle.enabled?(:schedule_virtual_hearings_for_central, user: current_user),
+  userCanAddVirtualHearingDays: FeatureToggle.enabled?(:national_vh_queue, user: current_user),
```
Comment (rubaiyat22) [inline app/views/hearings/index.html.erb:20]: does this feature toggle name make sense?

File: client/app/components/DateSelector.jsx [hunk 2]
```diff
@@ -85,7 +85,10 @@ DateSelector.propTypes = {
   /**
    * Text (or other node) to display to indicate an error state
    */
-  errorMessage: PropTypes.string,
+  errorMessage: PropTypes.oneOfType([
```
Comment (rubaiyat22) [inline client/app/components/DateSelector.jsx:88]: suppresses a warning on console

File: client/app/hearings/components/HearingDayAddModal.jsx [hunk 3]
```diff
@@ -57,310 +58,237 @@ const statusMsgDetailStyle = css({
...
-  getDateTypeErrorMessages = () => {
-    return <div>
-      <span {...statusMsgTitleStyle}>Cannot create a New Hearing Day</span>
-      <ul {...statusMsgDetailStyle} >
-        {
-          this.state.errorMessages.map((item, i) => <li key={i}>{item}</li>)
-        }
-      </ul></div>;
-  };
+  const videoHearingDateNotValid = (hearingDate) => {
```
Comment (rubaiyat22) [inline client/app/hearings/components/HearingDayAddModal.jsx:124]: I'm not sure this validation is necessary anymore

File: client/app/hearings/components/HearingDayAddModal.jsx [hunk 4]
```diff
@@ -374,9 +302,6 @@ HearingDayAddModal.propTypes = {
   onReceiveHearingSchedule: PropTypes.func,
   onRegionalOfficeChange: PropTypes.func,
   onSelectedHearingDayChange: PropTypes.func,
-  regionalOffices: PropTypes.shape({
```
Comment (rubaiyat22) [inline client/app/hearings/components/HearingDayAddModal.jsx:377]: unused prop

File: client/app/hearings/components/HearingDayAddModal.jsx [hunk 5]
```diff
@@ -393,17 +318,15 @@ HearingDayAddModal.propTypes = {
   selectedRegionalOffice: PropTypes.object,
 
   setNotes: PropTypes.func,
-  userCssId: PropTypes.string,
```
Comment (rubaiyat22) [inline client/app/hearings/components/HearingDayAddModal.jsx:396]: unused props

File: spec/feature/hearings/edit_hearing_day_spec.rb [hunk 6]
```diff
@@ -27,89 +23,108 @@
...
   end
 
-  scenario "first option is 'None'" do
-    visit "hearings/schedule"
-    find_link(hearing_day.scheduled_for.strftime("%a %-m/%d/%Y")).click
-    find("button", text: "Edit Hearing Day").click
-    find("label[for=roomEdit]").click
-    click_dropdown(name: "room", index: 0)
-    expect(dropdown_selected_value(find(".dropdown-room"))).to eq "None"
+  context "when request type is 'Virtual'" do
```
Comment (rubaiyat22) [inline spec/feature/hearings/edit_hearing_day_spec.rb:118]: new tests on this file

File: spec/feature/hearings/add_hearing_day_spec.rb [hunk 7]
```diff
@@ -155,4 +155,52 @@
       expect(page).to have_content(coordinator.full_name)
     end
   end
+
+  context "When adding a Virtual Hearing day" do
+    before { FeatureToggle.enable!(:national_vh_queue) }
```
Comment (tomas-nava) [inline spec/feature/hearings/add_hearing_day_spec.rb:160]:
```suggestion
    before { FeatureToggle.enable!(:national_vh_queue) }
    after { FeatureToggle.disable!(:national_vh_queue) }
```
I'm not sure this is necessary, but it's standard practice.

File: spec/feature/hearings/add_hearing_day_spec.rb [hunk 8]
```diff
@@ -155,4 +155,52 @@
...
+      find("button", text: "Add Hearing Date").click
+      expect(page).to have_content("Add Hearing Day")
+      click_dropdown(index: "R", text: "Virtual")
+      expect(page).to have_content("Select Regional Office (RO)", wait: 30)
+      fill_in "vlj", with: "Sallie L Anderson"
+      fill_in "coordinator", with: "Casimir R Funk"
+      fill_in "Notes (Optional)", with: "Test notes."
+      find("button", text: "Confirm").click
+      expect(page).to have_content("Please make sure you have entered a Hearing Date")
+    end
```
Comment (tomas-nava) [inline spec/feature/hearings/add_hearing_day_spec.rb:204]: you could combine these three tests into one, and use `step` to separate the logic, so

```ruby
    scenario "Adds a virtual hearing day" do
      step "Opening modal and selecting Virtual presents correct fields" do
        # same content as the same-named scenario
      end

      step "Leave Hearing Date without a selection, expect error" do
        # fill in everything except date & submit, check for error
      end

      step "Fill out all fields and confirm to save" do
        # fill in the date & submit, check for success
      end
    end
```

**Review by tomas-nava — APPROVED** [review]
approve!

With some non-blocking suggestions

**Pull request merged**